src/base/misc/disks.c

```diff
@@ -522,20 +523,24 @@ static void image_auto(struct disk *dp)
     dp->sectors = header.sectors;
     dp->tracks = header.cylinders;
     dp->header = header.header_end;
+    dp->num_secs = (unsigned long long)dp->tracks * dp->heads * dp->sectors;
   } else if (sect[510] == 0x55 && sect[511] == 0xaa) {
-    dp->tracks = 255;
+    filesize = lseek64(dp->fdesc, 0, SEEK_END);
+    dp->num_secs = (filesize + SECTOR_SIZE - 1) / SECTOR_SIZE;
     dp->heads = 255;
     dp->sectors = 63;
+    dp->tracks = (dp->num_secs + (dp->heads * dp->sectors - 1))
+		  / (dp->heads * dp->sectors);
```

Comment:
But why do you round-up this and "num_secs"
above? This will mean errors when an app tries
to access to the valid, in-range CHS values past
the image's end.

Reply:
Would you prefer it to extend the image to the next cylinder boundary?

Reply:
I prefer to truncate the CHS and, if you
insist, allow LBA w/o truncation (but still
truncate to the sector size!).
Accessing to the valid CHS values and
give "sector not found" just makes no
sense and can confuse disk diagnostic
tools.